app/controllers/secret_shares_controller.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class SecretSharesController < ApplicationController
+  before_action :ensure_secret_share_enabled!
+  skip_before_action :if_not_signed_in, only: [:show]
+
+  def ensure_secret_share_enabled!
```

Comment:
I think this should be a private method—it helps be clear about which methods are called by routes

Reply:
Ooh great suggestion @HashNotAdam! We typically put private methods at the bottom of the controller!